yt_dlp/extractor/lastfm.py

```diff
@@ -58,25 +34,17 @@ def _real_extract(self, url):
         return self.playlist_result(self._entries(url, playlist_id), playlist_id)
 
 
-class LastFMAlbumIE(LastFMPlaylistBaseIE):
-    _VALID_URL = r'https?://(?:www\.)?last\.fm/music/(?P<artist_id>[^/]+)/(?P<id>[^+/#?][^/#?]+)(/|[^/]*/?)$'
-    _TESTS = [{
-        'url': 'https://www.last.fm/music/Oasis/(What%27s+the+Story)+Morning+Glory%3F',
-        'info_dict': {
-            'id': '(What%27s+the+Story)+Morning+Glory%3F',
-        },
-        'playlist_count': 12,
-    }]
-
-
 class LastFMPlaylistIE(LastFMPlaylistBaseIE):
-    _VALID_URL = r'https?://(?:www\.)?last\.fm/(music|tag)/(?P<id>[^/#?]+)(/|/\+?tracks/?|[^/]*/?)$'
+    _VALID_URL = r'https?://(?:www\.)?last\.fm/(music|tag)/(?P<id>[^/]+)/?[^/]+?$'
```

Comment:
it should match
```
https://www.last.fm/music/Oasis/xxx
https://www.last.fm/music/Oasis
https://www.last.fm/music/Oasis/
https://www.last.fm/music/Oasis?ss
https://www.last.fm/music/Oasis#aa
https://www.last.fm/music/Oasis/?dd
https://www.last.fm/music/Oasis/xxx?aaa
```
but not
```
https://www.last.fm/music/Oasis/_/Wonderwall
```
right?
```suggestion
    _VALID_URL = r'https?://(?:www\.)?last\.fm/(music|tag)/(?P<id>[^/]+)(?:/[^/]+)?/?(?:[?#]|$)'
```